src/compiler/checker.ts

```diff
@@ -3090,15 +3090,24 @@ namespace ts {
             if (name.kind === SyntaxKind.Identifier) {
                 const message = meaning === namespaceMeaning || nodeIsSynthesized(name) ? Diagnostics.Cannot_find_namespace_0 : getCannotFindNameDiagnosticForName(getFirstIdentifier(name));
                 const symbolFromJSPrototype = isInJSFile(name) && !nodeIsSynthesized(name) ? resolveEntityNameFromAssignmentDeclaration(name, meaning) : undefined;
-                symbol = getMergedSymbol(resolveName(location || name, name.escapedText, meaning, ignoreErrors || symbolFromJSPrototype ? undefined : message, name, /*isUse*/ true));
+                let suggestedNameNotFoundMessage: DiagnosticMessage | undefined;
+                if (needSuggestedNameNotFoundMessage) {
+                    if (message === Diagnostics.Cannot_find_namespace_0) {
+                        suggestedNameNotFoundMessage = Diagnostics.Cannot_find_namespace_0_Did_you_mean_1;
+                    }
+                    else if (message === Diagnostics.Cannot_find_name_0) {
+                        suggestedNameNotFoundMessage = Diagnostics.Cannot_find_name_0_Did_you_mean_1;
+                    }
+                }
+                symbol = getMergedSymbol(resolveName(location || name, name.escapedText, meaning, ignoreErrors || symbolFromJSPrototype ? undefined : message, name, /*isUse*/ true, false, suggestedNameNotFoundMessage));

```

Comment:
`resolveName` now takes a boolean `issueSuggestions` instead of a message. I think it's probably fine to move your new code to choose the message inside `resolveName`.